query.go

```diff
@@ -187,7 +187,10 @@ func (q *query) constructLookupResult(target kb.ID) *lookupWithFollowupResult {
 	// determine if the query terminated early
 	completed := true
 
-	if !(q.isLookupTermination()) {
+	// Lookup and starvation are both valid ways for a lookup to complete. (Starvation does not imply failure.)
+	// Lookup termination (as defined in isLookupTermination) is not possible in small networks.
+	// Starvation is a successful query termination in small networks.
+	if !(q.isLookupTermination() || q.isStarvationTermination()) {
```

Comment:
Sure, it's a successful termination condition. However, perhaps we don't want to mark these queries as "completed" since that implies we know as much of the network as we're going to get out of this bucket. For networks small enough to naturally trigger the starvation condition it's probably very cheap to keep querying buckets, so maybe we should want to learn more about the network and not mark the query as "complete".